### PR TITLE
WIP: use deno to compile to single binary

### PR DESCRIPTION
**Pull request by 6543**
based on https://github.com/woodpecker-ci/plugin-ready-release-go/pull/609

```
 deno compile --allow-read --allow-write --allow-env --allow-net --allow-run src/run.ts
```

https://deno.com/blog/v1.46

### CODE REVIEW

**Comment by 6543** [comment]
image

well now i just need to make sure the git impl. is working currectly and cant be broken by user input ...

... but for my 1/2h experiment, it should be doable

File: src/utils/git.ts [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+  tags(options: string[]): Promise<any>;
+  log(options: any): Promise<any>;
+  merge(args: string[]): Promise<void>;
+  status(): Promise<{ isClean: () => boolean }>;
+  add(files: string | string[]): Promise<void>;
+  commit(message: string): Promise<void>;
+  push(args: string[]): Promise<void>;
+}
+
+export function simpleGit(): SimpleGit {
```
Comment (6543) [inline src/utils/git.ts:19]: mostly untested & we should add tests for it anyway ...

**Comment by pat-s** [comment]
Will this also reduce the image size? Could we even use a scratch image?

**Comment by 6543** [comment]
not scratch _jet_ as we need the git binary working

but yes that's a way to move towards it :)

**Comment by 6543** [comment]
the image size would probably increase by ~50Mb but the image would be silli and dont contain any unessesary build-time deps and other unrelated stuff

**Comment by 6543** [comment]
tldr: if we have a static linked git binary + the deno binary we should be able to use FROM scratch

**Comment by pat-s** [comment]
```
git clone git@github.com:git/git.git
mkdir git-static
cd git
make configure
./configure prefix=/path/to/git-static/ CFLAGS="${CFLAGS} -static"
make
make install
```

File: Dockerfile [hunk 2]
```diff
@@ -1,25 +1,31 @@
...
+# through /bin/sh (shelljs) and the forge API clients need CA certificates.
+# bookworm-slim is the smallest base that provides all of that.
+FROM debian:bookworm-slim
 WORKDIR /app
-ENV NODE_ENV=production
 
-RUN apt update \
-  && apt install -y git git-lfs wget \
+RUN apt-get update \
+  && apt-get install -y --no-install-recommends git git-lfs wget ca-certificates \
```
Comment (6543) [inline Dockerfile]:
```suggestion
RUN apt update \
  && apt install -y --no-install-recommends git git-lfs wget ca-certificates \
```
